src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeEventDispatcher.cs

```diff
@@ -179,6 +190,13 @@ private unsafe void DispatchEventsToEventListeners()
                 // Wait for more events.
                 if (!m_stopDispatchTask)
                 {
+                    if (!eventsReceived)
+                    {
+                        // Future TODO: this would make more sense to handle in EventPipeSession/EventPipe native code.
+                        Debug.Assert(!m_dispatchTaskWaitHandle.SafeWaitHandle.IsInvalid);
+                        m_dispatchTaskWaitHandle.WaitOne();
+                    }
+
                     Thread.Sleep(10);
```

Comment:
Do we still need this `sleep` here?  Doesn't seem to serve a purpose now if I'm reading this loop correctly.

Reply:
It's not functionally required, but the idea is to give a buffer zone so that we batch up a reasonable number of events before converting the write buffer to read only. Without it we could be potentially creating a huge buffer for one or two events.

It's still possible with it if the events are coming in slowly but it's acceptable it only happens once every 100 ms.